### PR TITLE
Email testing with Mailhog

### PR DESCRIPTION
**Pull request by AronNovak**
#153

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -5,7 +5,7 @@
 <!-- Using this project's bootstrap file allows tests in `ExistingSite`,
     `ExistingSiteSelenium2DriverTest`, and `ExistingSiteWebDriverTest`
      to run alongside core's test types. -->
-<phpunit bootstrap="vendor/weitzman/drupal-test-traits/src/bootstrap-fast.php">
+<phpunit bootstrap="vendor/weitzman/drupal-test-traits/src/bootstrap.php">
```
Comment (amitaibu) [inline phpunit.xml.dist:8]: Let's keep `fast`. Did you get an error?

File: web/modules/custom/server_general/tests/src/Traits/ServerGeneralMailTrait.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+trait ServerGeneralMailTrait {
+
+  /**
+   * Mailhog base URL as provided by DDEV.
+   *
+   * @return string
+   *   Fully qualified URL of Mailhog.
+   */
+  public function getMailhogBaseUrl() {
+    return 'https://drupal-starter.ddev.site:8026';
```
Comment (amitaibu) [inline web/modules/custom/server_general/tests/src/Traits/ServerGeneralMailTrait.php:17]: I wonder if we could load it from the ENV

File: phpunit.xml.dist (hunk 1 above)
Comment (AronNovak) [reply phpunit.xml.dist:8]: I did. Traits are not recognized.

File: web/modules/custom/server_general/tests/src/Traits/ServerGeneralMailTrait.php (hunk 2 above)
Comment (AronNovak) [reply web/modules/custom/server_general/tests/src/Traits/ServerGeneralMailTrait.php:17]: I'll try, good point!

File: phpunit.xml.dist (hunk 1 above)
Comment (amitaibu) [reply phpunit.xml.dist:8]: https://gitlab.com/weitzman/drupal-test-traits/-/blob/master/src/bootstrap-fast.php#L8-12

Comment (AronNovak) [reply phpunit.xml.dist]: nice, it was quite easy!

File: web/modules/custom/server_general/tests/src/Traits/ServerGeneralMailTrait.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Drupal\Tests\server_general\Traits;
+
+/**
+ * Email testing trait.
+ */
+trait ServerGeneralMailTrait {
```
Comment (amitaibu) [inline web/modules/custom/server_general/tests/src/Traits/ServerGeneralMailTrait.php:8]:
```suggestion
trait ServerGeneralMailTestTrait {
```

File: web/modules/custom/server_general/tests/src/ExistingSite/ServerGeneralUserTest.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace Drupal\Tests\server_general\ExistingSite;
+
+use Drupal\Tests\server_general\Traits\ServerGeneralMailTrait;
+use weitzman\DrupalTestTraits\ExistingSiteBase;
+
+/**
+ * A model test case for email-testing using traits from Drupal Test Traits.
+ */
+class ServerGeneralUserTest extends ExistingSiteBase {
```
Comment (amitaibu) [inline web/modules/custom/server_general/tests/src/ExistingSite/ServerGeneralUserTest.php:11]: Maybe

```suggestion
class ServerGeneralMailTest extends ExistingSiteBase {
```

File: web/modules/custom/server_general/tests/src/ExistingSite/ServerGeneralMailTest.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+   */
+  public function testOneTimeLoginLinkEmail() {
+    $this->resetOutgoingMails();
+    $this->assertOutgoingMailNumber(0);
+    $this->drupalGet('user/password');
+    $this->getCurrentPage()->fillField('edit-name', 'joe@example.com');
+    $this->getCurrentPage()->pressButton('Submit');
+    $this->assertOutgoingMailNumber(1);
+    $this->assertOutgoingMailContains('Replacement login information for JoeDoe at Drupal Starter');
+
```
Comment (AronNovak) [inline web/modules/custom/server_general/tests/src/ExistingSite/ServerGeneralMailTest.php:30]:
```suggestion
```

**Review by amitaibu — APPROVED** [review]
:ok_hand:

**Pull request merged**